### DISCUSSION TITLE
Highlight specific listchars

### DISCUSSION THREAD

**Discussion by avidseeker** [vim/vim#15439] [Q&A]
It seems that there's only one syntax highlighting group for `listchars` that applies a style to all of them.

How can I highlight `extends,precedes` in different colors than whitespace e.g: `tab,eol,nbsp`?

**Comment by georgalis** [comment]
Try looking at this as a highlighting challenge not a listchars challenge.
- define a custom highlight group for each of your new colored styles
- define a match pattern (after defining the highlight group)
```
highlight CustomTab ctermbg=red guibg=red
match CustomTab /\t/
```
I think the match regex applies after any listchars modification, so the regex should align with your config. Review `:help match` and `:help 3match` I think the latter applies a precedence in cases where multiple regex match.

It would certainly be convenient to have a predefined highlight name and regex match for each of the listchar string settings, but I think that would be a feature request.

**Reply by avidseeker** [reply]
This would highlight all tabs whether `:set list` or `:set nolist`. I need to only change the color of tabs rendered by substituted symbols in `listchars`. When a tab is not replaced by symbols, don't apply any specific highlighting.

**Comment by georgalis** [comment]
right, you will need to craft a regex to your exact needs (the listchars modification). you could conditionally apply custom highlight when list is set but I am not interested in figuring that out. as mentioned the listchar string settings do not have corresponding syntax highlight.
